src/wp-includes/post.php

```diff
@@ -3400,21 +3400,45 @@ function wp_count_posts( $type = 'post', $perm = '' ) {
 		return apply_filters( 'wp_count_posts', $counts, $type, $perm );
 	}
 
-	$query = "SELECT post_status, COUNT( * ) AS num_posts FROM {$wpdb->posts} WHERE post_type = %s";
+	$query = $wpdb->prepare( "SELECT post_status, COUNT(*) AS num_posts FROM {$wpdb->posts} WHERE post_type = %s", $type );
```

Comment:
If this `$query` is just getting overridden in the nested `if` statement below now, should, it seems like this can be set in an `else` statement.